core/src/components/footer/footer.tsx

```diff
@@ -12,30 +18,18 @@ export class TdsFooter {
   /** Mode variant of the component, based on current mode. */
   @Prop() modeVariant: 'primary' | 'secondary' = null;
 
-  /** Indicates whether the Footer has a top part. */
-  @State() hasTopPart: boolean;
-
-  connectedCallback() {
-    this.hasTopPart = Array.from(this.host.children).some((element) => element.slot === 'top');
-  }
-
   render() {
+    const usesTopSlot = hasSlot('top', this.host);
+    const usesStartSlot = hasSlot('start', this.host);
+    const usesEndSlot = hasSlot('end', this.host);
     return (
       <Host class={`${this.modeVariant ? `tds-mode-variant-${this.modeVariant}` : ''}`}>
         <footer>
-          {this.hasTopPart && (
-            <div class="footer-top">
-              <slot name="top"></slot>
-            </div>
-          )}
+          {usesTopSlot && <slot name="top"></slot>}
           <div class="footer-main">
             <div class="footer-main-top">
-              <div class="footer-bottom-left">
-                <slot name="main-left"></slot>
-              </div>
-              <div class="footer-bottom-right">
-                <slot name="main-right"></slot>
-              </div>
+              {usesStartSlot && <slot name="start"></slot>}
```

Comment:
I might be misunderstanding here, but why are we not documenting 'start' as a slot together with 'left', 'right' and 'top'?

Reply:
Not sure actaully... Haha. I think I was sort of uncertain on the naming actaully

Reply:
Updated now!